### PR TITLE
[C/C++] syntax improvement: numeric constants detection improvement

### PR DESCRIPTION
**Pull request by dannftk**
1. bring out numeric constants in variables section
2. preprocessor numeric constants detection improvement

### CODE REVIEW

**Comment by dannftk** [comment]
How to request someone to run CI on the PR?

**Comment by rwols** [comment]
> How to request someone to run CI on the PR?

Some maintainer needs to click a button.

Or: a maintainer can edit the repo settings to automatically run the CI for users whose account is older than X days. It is currently disabled for all first-time contributors, which is too strict.

File: C++/C++.sublime-syntax [hunk 1]
```diff
@@ -205,31 +210,31 @@ contexts:
     # https://en.cppreference.com/w/cpp/language/integer_literal
 
     # hexadecimal integer
-    - match: \b(0[xX])({{hex_digit}}*)({{hex_suffix}})?
+    - match: '{{hex_integer}}'
```
Comment (deathaxe) [inline C++/C++.sublime-syntax:213]: By moving those patterns to variables its just harder to keep track of capture groups.

The only reason to introduce variables was to provide an easier way for an inherited syntax to tweak those patterns without touching contexts. 

But I guess that's not scope of this PR.

**Review by deathaxe — CHANGES_REQUESTED** [review]
Would you mind adding some test cases which cover your code changes, please, so we can make sure to not break those by possible future changes?

**Review by deathaxe — APPROVED** [review]

**Comment by dannftk** [comment]
How do I request one more reviewer to glance at the PR?

**Review by jfcherng — APPROVED** [review]

**Pull request merged**